checkov/bicep/runner.py

```diff
@@ -1,57 +1,105 @@
 from __future__ import annotations
 
-import re
+import logging
 from pathlib import Path
-from typing import cast
+from typing import cast, Type, TYPE_CHECKING
 
+from pycep.typing import BicepJson
 from typing_extensions import Literal
 
 from checkov.bicep.checks.param.registry import registry as param_registry
 from checkov.bicep.checks.resource.registry import registry as resource_registry
+from checkov.bicep.graph_builder.local_graph import BicepLocalGraph
+from checkov.bicep.graph_manager import BicepGraphManager
 from checkov.bicep.parser import Parser
-from checkov.common.checks.base_check_registry import BaseCheckRegistry
-from checkov.common.comment.enum import COMMENT_REGEX
-from checkov.common.models.enums import CheckResult
+from checkov.bicep.utils import clean_file_path, get_scannable_file_paths, search_for_suppression
+from checkov.common.checks_infra.registry import get_graph_checks_registry
+
+from checkov.common.graph.db_connectors.networkx.networkx_db_connector import NetworkxConnector
+from checkov.common.graph.graph_builder import CustomAttributes
+from checkov.common.output.graph_record import GraphRecord
 from checkov.common.output.record import Record
 from checkov.common.output.report import CheckType, Report
 from checkov.common.runners.base_runner import BaseRunner
 from checkov.common.typing import _CheckResult
 from checkov.runner_filter import RunnerFilter
 
+if TYPE_CHECKING:
```

Comment:
@gruebel Why is this `if` needed?

Reply:
if you add `if TYPE_CHECKING:` and under it different imports, they will be only used for type hint checking, so they are not loaded during runtime, which has of course some performance benefits, not so important here, because those classes are loaded anyway somewhere else in the code, but it circumvents accidentally creating circular dependencies, when you actually only wanted to add type hints.
This is the official way of doing it, and can be found in other open source projects with a lot of type hints.